pkg/dependencycheck/dependencycheck.go

```diff
@@ -0,0 +1,37 @@
+package dependencycheck
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+// DependencyCheck implements the interfaces to fetch vulnerabilities and solutions from nexpose
+type DependencyCheck struct {
+	HTTPClient  *http.Client
+	NexposeHost *url.URL
+}
+
+// DepCheck fetches the solutions to a particular vulnerability
+func (dc *DependencyCheck) DepCheck(ctx context.Context) error {
+	u, _ := url.Parse(dc.NexposeHost.String() + "/api/3/solutions/mysql-upgrade-latest")
+	req, _ := http.NewRequest(http.MethodGet, u.String(), http.NoBody)
+	res, err := dc.HTTPClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != 200 {
+		// Nexpose is down?  Nexpose unexpectedly gave 404?
+		headers, errMarshal := json.Marshal(res.Header)
```

Comment:
question: Why are we logging response headers? What information are we attempting to capture?

Reply:
yea that was some copy pasta, i'll remove it